zcash_primitives/src/transaction/mod.rs

```diff
@@ -370,6 +370,68 @@ impl<A: Authorization> TransactionData<A> {
             digester.digest_tze(self.tze_bundle.as_ref()),
         )
     }
+
+    /// Maps the bundles from one type to another.
+    ///
+    /// This shouldn't be necessary for most use cases; it is provided for handling the
+    /// cross-FFI builder logic in `zcashd`.
+    pub fn map_bundles<B: Authorization>(
+        self,
+        f_transparent: impl FnOnce(
+            Option<transparent::Bundle<A::TransparentAuth>>,
+        ) -> Option<transparent::Bundle<B::TransparentAuth>>,
+        f_sapling: impl FnOnce(
+            Option<sapling::Bundle<A::SaplingAuth>>,
+        ) -> Option<sapling::Bundle<B::SaplingAuth>>,
+        f_orchard: impl FnOnce(
+            Option<orchard::bundle::Bundle<A::OrchardAuth, Amount>>,
+        ) -> Option<orchard::bundle::Bundle<B::OrchardAuth, Amount>>,
+        #[cfg(feature = "zfuture")] f_tze: impl FnOnce(
+            Option<tze::Bundle<A::TzeAuth>>,
+        ) -> Option<tze::Bundle<B::TzeAuth>>,
+    ) -> TransactionData<B> {
+        TransactionData {
+            version: self.version,
+            consensus_branch_id: self.consensus_branch_id,
+            lock_time: self.lock_time,
+            expiry_height: self.expiry_height,
+            transparent_bundle: f_transparent(self.transparent_bundle),
+            sprout_bundle: self.sprout_bundle,
+            sapling_bundle: f_sapling(self.sapling_bundle),
+            orchard_bundle: f_orchard(self.orchard_bundle),
+            #[cfg(feature = "zfuture")]
+            tze_bundle: f_tze(self.tze_bundle),
+        }
+    }
+
+    pub fn map_authorization<B: Authorization>(
```

Comment:
I think this one is technically not needed anymore now that `TransactionData::map_bundles` exists? But I don't mind it being here.